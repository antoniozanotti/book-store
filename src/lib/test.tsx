import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
