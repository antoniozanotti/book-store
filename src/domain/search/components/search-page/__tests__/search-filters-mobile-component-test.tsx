import { render, screen } from "@/lib/test";
import { SearchFiltersMobileComponent } from "../search-filters-mobile-component";

describe("SearchFiltersMobileComponent", () => {
  it("should render", async () => {
    const { container } = render(<SearchFiltersMobileComponent />);
    expect(container).toMatchSnapshot();
  });
});