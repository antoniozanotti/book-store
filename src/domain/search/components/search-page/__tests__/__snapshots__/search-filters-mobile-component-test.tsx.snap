// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFiltersMobileComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <div
      class="px-4 lg:hidden bg-primary-100"
    >
      <h2
        class="font-primary text-pretty text-2xl hidden lg:block p-10 pb-0"
      >
        Filter
      </h2>
      <h2
        class="font-primary text-pretty text-2xl hidden lg:block p-10 pb-0"
      >
        Order by
      </h2>
      <div
        class="w-full px-0"
        data-orientation="vertical"
      >
        <div
          aria-label="Search Options"
          class="group-[.is-splitted]:px-4 group-[.is-splitted]:bg-content1 group-[.is-splitted]:shadow-medium group-[.is-splitted]:rounded-medium"
        >
          <h2>
            <button
              aria-expanded="false"
              class="flex py-4 w-full h-full gap-3 items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 transition-opacity"
              id=":r0:"
              type="button"
            >
              <div
                class="flex-1 flex flex-col text-start"
              >
                <span
                  class="text-foreground text-large"
                >
                  <h2
                    class="font-primary text-pretty text-2xl"
                  >
                    Search Options
                  </h2>
                </span>
              </div>
              <span
                aria-hidden="true"
                class="text-default-400 transition-transform rotate-0 data-[open=true]:-rotate-90 rtl:-rotate-180 rtl:data-[open=true]:-rotate-90"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M15.5 19l-7-7 7-7"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
            </button>
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
